pkg/k8s/apis/cilium.io/v2/register.go

```diff
@@ -128,17 +148,57 @@ func CreateCustomResourceDefinitions(clientset apiextensionsclient.Interface) er
 			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
 				Plural:     CustomResourceDefinitionPluralName,
 				Singular:   CustomResourceDefinitionSingularName,
-				ShortNames: []string{"cnp", "ciliumnp"},
+				ShortNames: CustomResourceDefinitionShortNames,
 				Kind:       CustomResourceDefinitionKind,
 			},
 			Scope:      apiextensionsv1beta1.NamespaceScoped,
-			Validation: &crv,
+			Validation: &cnpCRV,
 		},
 	}
 
 	return createUpdateCRD(clientset, "CiliumNetworkPolicy/v2", res)
 }
 
+// createCEPCRD creates and updates the CiliumEndpoint CRD. It should be called
+// on agent startup but is idempotent and safe to call again.
+func createCEPCRD(clientset apiextensionsclient.Interface) error {
+	var (
+		// CustomResourceDefinitionSingularName is the singular name of custom resource definition
+		CustomResourceDefinitionSingularName = "ciliumendpoint"
+
+		// CustomResourceDefinitionPluralName is the plural name of custom resource definition
+		CustomResourceDefinitionPluralName = "ciliumendpoints"
+
+		// CustomResourceDefinitionShortNames are the abbreviated names to refer to this CRD's instances
+		CustomResourceDefinitionShortNames = []string{"cep"}
```

Comment:
`cep` was slightly confusing to me in the beginning as I was mapping it to `Endpoints` and not `Endpoint` but I can't think of anything better as `ce` is likely too short.

Reply:
yeah :/  I was following the cnp precedent (it also has ciliumnp but ciliumep seemed a little off). I could have `cep` and `ceps`.
I'm also open to other names, especially if they're more obviously not related to the k8s endpoints. On the other hand, we already call them endpoints so :(

Reply:
yah, sure. I'll add it.